sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientTests.cs

```diff
@@ -1475,6 +1475,436 @@ public async Task ProcessingStartsWithTheExactDefaultPositionWithNoCheckpoint()
+        ///   and <see cref="EventProcessorClient.StopProcessingAsync(CancellationToken)" /> methods.
+        /// </summary>
+        ///
+        [Test]
+        public async Task SupportsStartProcessingAfterStop()
+        {
+            var mockConsumer = new Mock<EventHubConsumerClient>("consumerGroup", Mock.Of<EventHubConnection>(), default);
+            var mockProcessor = new Mock<EventProcessorClient>(Mock.Of<PartitionManager>(), "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), default) { CallBase = true };
+
+            mockConsumer
+                .Setup(consumer => consumer.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .Returns(Task.FromResult(Array.Empty<string>()));
+
+            mockProcessor
+                .Setup(processor => processor.CreateConsumer(
+                    It.IsAny<string>(),
+                    It.IsAny<EventHubConnection>(),
+                    It.IsAny<EventHubConsumerClientOptions>()))
+                .Returns(mockConsumer.Object);
+
+            mockProcessor.Object.ProcessEventAsync += eventArgs => Task.CompletedTask;
+            mockProcessor.Object.ProcessErrorAsync += eventArgs => Task.CompletedTask;
+
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+            Assert.That(mockProcessor.Object.IsRunning, Is.False);
+
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
+
+            Assert.That(mockProcessor.Object.IsRunning, Is.True);
+
+            await mockProcessor.Object.StopProcessingAsync(cancellationSource.Token);
+
+            Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The processor should have stopped without cancellation.");
+            Assert.That(mockProcessor.Object.IsRunning, Is.False);
+
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
+            Assert.That(mockProcessor.Object.IsRunning, Is.True);
+
+            await mockProcessor.Object.StopProcessingAsync(cancellationSource.Token);
+
+            Assert.That(mockProcessor.Object.IsRunning, Is.False);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorClient.StopProcessingAsync(CancellationToken)" /> method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task StopProcessingShouldSurfaceLoadBalancingException()
+        {
+            var mockProcessor = new Mock<EventProcessorClient>(Mock.Of<PartitionManager>(), "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), default) { CallBase = true };
+            var completionSource = new TaskCompletionSource<bool>();
+
+            mockProcessor
+                .Setup(processor => processor.CreateConsumer(
```

Comment:
I'm a bit confused by this.  Wouldn't we expect these to surface via the `ProcessErrors` handler?   Shouldn't this be one of those cases where we give up ownership of a partition but keep on going?

@kinelski :  You may be the best source of context here.

Reply:
@ShivangiReja :  Don't let this block you;  nothing wrong with the code here.   I'll reflect this question into chat for when Caio returns.

Reply:
There are two scenarios in which we call the `CreateConsumer` method:
- When reading events from an owned partition: this happens in a separate process and works exactly as you're expecting. In case of failure, the error handler is called.
- In the `RunAsync` method, in order to retrieve the list of partition ids from the Event Hubs service: this one is called only once, right before entering the load balancing loop. This occurrence is responsible for the processor failure.

https://github.com/Azure/azure-sdk-for-net/blob/8c8c8c62704c23cc2044c868de5a8c2fed407865/sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs#L833

Do you think it should behave differently?

Reply:
Nope.   That makes sense.   I was missing the second case.  Thanks!